pkg/apis/k0s.k0sproject.io/v1beta1/clusterconfig_types.go

```diff
@@ -218,10 +219,10 @@ func (c *ClusterConfig) UnmarshalJSON(data []byte) error {
 	type config ClusterConfig
 	jc := (*config)(c)
 
-	if err := json.Unmarshal(data, jc); err != nil {
-		return err
-	}
-	return nil
+	decoder := json.NewDecoder(bytes.NewReader(data))
+	decoder.DisallowUnknownFields()
```

Comment:
the yaml package already turns `DisallowUnknownFields` on. Why isn't it working then?
https://github.com/kubernetes-sigs/yaml/blob/39f74b9b8a135185d095704ab98c760ca031a9f8/yaml.go#L64

Reply:
Yes. The `json.Unmarshal` in the `UnmarshalJSON` function creates a new decoder that doesn't know anything about the `DisallowUnknownFields` set by the yaml thing.